File: src/network/packet/impl/PlayerUpdateNotificationStatePacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\cache\NotificationListCache;
use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;

final class PlayerUpdateNotificationStatePacket extends CloudPacket implements CloudboundPacket {

    public function __construct(
        private string $player = "",
        private bool $value = false
    ) {}

    public function handle(ServerClient $client): void {
        if ($this->value) NotificationListCache::add($this->player);
        else NotificationListCache::remove($this->player);
    }

    public function encodePayload(PacketData $packetData): void {}

    public function decodePayload(PacketData $packetData): void {
        $packetData->readAll($this->player, $this->value);
    }

    public function getPlayer(): string {
        return $this->player;
    }

    public function isValue(): bool {
        return $this->value;
    }

    public static function create(string $player, bool $value): self {
        return new self($player, $value);
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/network/packet/impl/PlayerUpdateNotificationStatePacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.network.packet.AuthenticatedPacket;
import de.pocketcloud.network.packet.CloudboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.network.packet.data.PacketData;
import de.pocketcloud.cloud.provider.CloudProvider;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

@NoArgsConstructor
@Getter
public final class PlayerUpdateNotificationStatePacket extends CloudPacket implements CloudboundPacket, AuthenticatedPacket {

    private String player;
    private boolean value;

    public PlayerUpdateNotificationStatePacket(String player, boolean value) {
        this.player = player != null ? player : "";
        this.value = value;
    }

    @Override
    public void handle(@NotNull ServerClient client) {
        if (value) CloudProvider.current().enablePlayerNotifications(player);
        else CloudProvider.current().disablePlayerNotifications(player);
    }

    @Override
    public void encodePayload(PacketData packetData) {}

    @Override
    public void decodePayload(PacketData packetData) {
        this.player = packetData.readString();
        this.value = packetData.readBool();
    }

    public static PlayerUpdateNotificationStatePacket create(String player, boolean value) {
        return new PlayerUpdateNotificationStatePacket(player, value);
    }
}
